### PR TITLE
Document jslib-aws v0.12.1

### PR DESCRIPTION
**Pull request by oleiade**
## What?

This Pull Request updates the jslib/aws documentation according to the changes made in v0.12.0 that we've just published.

Furthermore, I also went ahead and fixed the documentation and examples of the whole SignatureV4 API to address the recent undocumented changes we made to it.
 
## Checklist

- [X] I have used a meaningful title for the PR.
- [X] I have described the changes I've made in the "What?" section above.
- [X] I have performed a self-review of my changes.
- [X] I have run the `npm start` command locally and verified that the changes look good.
- [X] I have made my changes in the `docs/sources/next` folder of the documentation.
- [X] I have reflected my changes in the `docs/sources/v{most_recent_release}` folder of the documentation.
- [X] I have reflected my changes in the relevant folders of the two previous k6 versions of the documentation (if still applicable to previous versions).

## Related PR(s)/Issue(s)

https://github.com/grafana/k6-jslib-aws/releases/tag/v0.12.0

https://github.com/grafana/k6-jslib-aws/issues/102
https://github.com/grafana/k6-jslib-aws/issues/103
https://github.com/grafana/k6-jslib-aws/issues/67

### CODE REVIEW

**Comment by oleiade** [comment]
I've published jslib-aws v0.12.1 today, and updated the PR to reflect this change 👍🏻

**Comment by oleiade** [comment]
Up 👀 @heitortsergent, can I get your eyes on this, please? The current documentation is outdated 🙂

File: docs/sources/next/javascript-api/jslib/aws/LambdaClient/_index.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+head_title: 'LambdaClient'
+description: 'LambdaClient allows interacting with the AWS Lambda service'
+weight: 00
+---
+
+# LambdaClient
+
+{{< docs/shared source="k6" lookup="blocking-aws-blockquote.md" version="<K6_VERSION>" >}}
+
+`LambdaClient` interacts with the AWS Lambda. With it, you can invoke a lambda function.
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/jslib/aws/LambdaClient/_index.md:12]:
```suggestion
`LambdaClient` interacts with [AWS Lambda](https://aws.amazon.com/lambda/). With it, you can invoke a Lambda function.
```

File: docs/sources/next/javascript-api/jslib/aws/LambdaClient/_index.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+Both the dedicated `lambda.js` jslib bundle and the all-encompassing `aws.js` bundle include the `LambdaClient`.
+
+### Methods
+
+| Function                                                                                                                  | Description                     |
+| :------------------------------------------------------------------------------------------------------------------------ | :------------------------------ |
+| [invoke(name, payload, [options])](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/jslib/aws/lambdaclient/invoke) | Invokes an AWS lambda function. |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/jslib/aws/LambdaClient/_index.md:20]:
```suggestion
| [invoke(name, payload, [options])](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/jslib/aws/lambdaclient/invoke) | Invokes an AWS Lambda function. |
```

File: docs/sources/next/javascript-api/jslib/aws/LambdaClient/_index.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+### Throws
+
+`LambdaClient` methods will throw errors in case of failure.
+
+| Error                 | Condition                                                   |
+| :-------------------- | :---------------------------------------------------------- |
+| InvalidSignatureError | when invalid credentials were provided.                     |
+| LambdaInvocationError | when AWS replied to the requested invokation with an error. |
+
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/jslib/aws/LambdaClient/_index.md:30]:
```suggestion
| InvalidSignatureError | When invalid credentials were provided.                     |
| LambdaInvocationError | When AWS replied to the requested invocation with an error. |

```

File: docs/sources/next/javascript-api/jslib/aws/LambdaClient/invoke.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
+---
+title: 'invoke'
+head_title: 'LambdaClient.invoke(name, payload, [options])'
+description: 'LambdaClient.deleteObject deletes an object from a bucket'
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/jslib/aws/LambdaClient/invoke.md:4]:
```suggestion
description: 'LambdaClient.invoke invokes an AWS Lamba function'
```

File: docs/sources/next/javascript-api/jslib/aws/LambdaClient/invoke.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+| Parameter | Type                                    | Description                                     |
+| :-------- | :-------------------------------------- | :---------------------------------------------- |
+| name      | string                                  | Name of the function to invoke.                 |
+| payload   | string                                  | Payload to send to the function being invoked.  |
+| options   | [InvocationOptions](#invocationoptions) | Additional options to customize the invocation. |
+
+#### InvocationOptions
+
+InvocationOptions is an object used to customize the invocation of a lambda function.
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/jslib/aws/LambdaClient/invoke.md:22]:
```suggestion
InvocationOptions is an object used to customize the invocation of a Lambda function.
```

File: docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/_index.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+head_title: 'LambdaClient'
+description: 'LambdaClient allows interacting with the AWS Lambda service'
+weight: 00
+---
+
+# LambdaClient
+
+{{< docs/shared source="k6" lookup="blocking-aws-blockquote.md" version="<K6_VERSION>" >}}
+
+`LambdaClient` interacts with the AWS Lambda. With it, you can invoke a lambda function.
```
Comment (heitortsergent) [inline docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/_index.md:12]:
```suggestion
`LambdaClient` interacts with [AWS Lambda](https://aws.amazon.com/lambda/). With it, you can invoke a Lambda function.
```

File: docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/_index.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+Both the dedicated `lambda.js` jslib bundle and the all-encompassing `aws.js` bundle include the `LambdaClient`.
+
+### Methods
+
+| Function                                                                                                                  | Description                     |
+| :------------------------------------------------------------------------------------------------------------------------ | :------------------------------ |
+| [invoke(name, payload, [options])](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/jslib/aws/lambdaclient/invoke) | Invokes an AWS lambda function. |
```
Comment (heitortsergent) [inline docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/_index.md:20]:
```suggestion
| [invoke(name, payload, [options])](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/jslib/aws/lambdaclient/invoke) | Invokes an AWS Lambda function. |
```

File: docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/_index.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+### Throws
+
+`LambdaClient` methods will throw errors in case of failure.
+
+| Error                 | Condition                                                   |
+| :-------------------- | :---------------------------------------------------------- |
+| InvalidSignatureError | when invalid credentials were provided.                     |
+| LambdaInvocationError | when AWS replied to the requested invokation with an error. |
```
Comment (heitortsergent) [inline docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/_index.md:29]:
```suggestion
| InvalidSignatureError | When invalid credentials were provided.                     |
| LambdaInvocationError | When AWS replied to the requested invokation with an error. |
```

File: docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/invoke.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
+---
+title: 'invoke'
+head_title: 'LambdaClient.invoke(name, payload, [options])'
+description: 'LambdaClient.deleteObject deletes an object from a bucket'
```
Comment (heitortsergent) [inline docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/invoke.md:4]:
```suggestion
description: 'LambdaClient.invoke invokes an AWS Lamba function'
```

File: docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/invoke.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+| Parameter | Type                                    | Description                                     |
+| :-------- | :-------------------------------------- | :---------------------------------------------- |
+| name      | string                                  | Name of the function to invoke.                 |
+| payload   | string                                  | Payload to send to the function being invoked.  |
+| options   | [InvocationOptions](#invocationoptions) | Additional options to customize the invocation. |
+
+#### InvocationOptions
+
+InvocationOptions is an object used to customize the invocation of a lambda function.
```
Comment (heitortsergent) [inline docs/sources/v0.51.x/javascript-api/jslib/aws/LambdaClient/invoke.md:22]:
```suggestion
InvocationOptions is an object used to customize the invocation of a Lambda function.
```

**Review by heitortsergent — APPROVED** [review]
Just some small nits, LGTM overall! Thanks for your patience @oleiade 🙇 .

**Pull request merged**